ovgs.proto

```diff
@@ -273,85 +276,86 @@ service OwnershipVoucherService {
+  // GetSerial returns component, groups the component belongs to.
   // Errors will be returned:
-  // INVALID_ARGUMENT if the serial number is empty or the IEN supplied
-  // isn't applicable for the voucher issuer
-  // NOT_FOUND if the serial number doesn't exist.
+  // INVALID_ARGUMENT if any field of component or group_id is empty or the
+  // IEN isn't applicable for the voucher issuer.
+  // NOT_FOUND if the component doesn't exist.
   // PERMISSION_DENIED if the user doesn't have access to the group
   // Roles with permission to invoke this = ADMIN, ASSIGNER, REQUESTOR
   rpc GetSerial(GetSerialRequest) returns (GetSerialResponse);
 
   // CreateDomainCert creates the certificate in the group.
   // Errors will be returned:
-  // NOT_FOUND if the group doesn't exist, as specified in the request
-  // INVALID_ARGUMENT if expiry_time is empty or in the past or the
-  // suuplied cert is invalid
-  // ALREADY_EXISTS if cert already exists in the group
+  // INVALID_ARGUMENT if expiry_time is empty or in the past, the
+  // supplied cert is invalid (such expired or malformed), or any of the
+  // fields are empty.
+  // NOT_FOUND if the group_id doesn't exist.
+  // ALREADY_EXISTS if the certificate_der,revocation_checks,expiry_time
+  // tuple already exists in the group.
   // PERMISSION_DENIED if the user doesn't have access to the group
   // Roles with permission to invoke this = ADMIN, ASSIGNER
   rpc CreateDomainCert(CreateDomainCertRequest) returns (CreateDomainCertResponse);
-  // DeleteDomainCert deletes the cert from the database.
+  // DeleteDomainCert deletes the cert_id.
   // Errors will be returned:
-  // NOT_FOUND if the cert doesn't exist
+  // NOT_FOUND if the cert_id doesn't exist
   // PERMISSION_DENIED if user doesn't have access to the group
   // Roles with permission to invoke this = ADMIN, ASSIGNER
   rpc DeleteDomainCert(DeleteDomainCertRequest) returns (DeleteDomainCertResponse);
-  // GetDomainCert reveals the details of the certificate.
-  // NOT_FOUND if the cert doesn't exist
+  // GetDomainCert returns the details of the cert_id.
+  // NOT_FOUND if the cert_id doesn't exist.
   // PERMISSION_DENIED if user doesn't have access to the group
   // Roles with permission to invoke this = ADMIN, ASSIGNER, REQUESTOR
   rpc GetDomainCert(GetDomainCertRequest) returns (GetDomainCertResponse);
 
-  // GetOwnershipVoucher issues ownership voucher for the serial number (if it
+  // GetOwnershipVoucher issues an ownership voucher for the component (if it
   // exists/if applicable)
   // Errors will be returned:
-  // INVALID_ARGUMENT if serial number, IEN, cert_id, or lifetime are empty,
-  // lifetime is in the past or the IEN supplied isn't applicable for the voucher
-  // issuer
+  // INVALID_ARGUMENT if any field of the request is empty, lifetime is in
+  // the past, or the IEN supplied isn't applicable for the voucher issuer
+  // FAILED_PRECONDITION if the component or cert_id do not exist.
```

Comment:
Why the failed_precondition on the component or cert_id not existing?

A simple NOT_FOUND seems consistent with how the absence of things is handled in places above

Reply:
I just based this on the other use of failed-precondition in the proto, where AddUserRole uses it for missing user or group.  I guessed that you had used it because those were external from the operation, in a sense; things that had to be added separately.

I suppose cert_id should be failed-precondition, and component is not found.  But, I really do not care.  Tell me what you prefer.

Reply:
NOT_FOUND works